app/controllers/vouchers_controller.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class VouchersController < ApplicationController
+  before_action :only => [:show] do
```

Comment:
So users can list them, edit them and create them, but can not get details of a single voucher?